wgsl/index.bs

```diff
@@ -3511,12 +3520,12 @@ The result type depends on the number of letters provided. Assuming a `vec4<f32>
            |e|[|i|] : |T|
        <td>Select the |i|'<sup>th</sup> component of vector<br>
            The first component is at index |i|=0.<br>
-           If |i| is outside the range [0,|N|-1], then an index in the range [0, |N|-1] is used instead.
+           If |i| is outside the range [0,|N|-1], then expression evaluates to either:
```

Comment:
same concern about RBA

Reply:
This is not a memory value so I'm not sure that is appropriate. The vector has been loaded already here and this is then extracting an out of bounds index.

Reply:
True. So what does Vulkan/SPIR-V say is going to happen in this case?

Reply:
It produces an undefined value, but I do think we could opt for more restrictive behaviour that is written here (by using a max/clamp operation).

Reply:
We should discuss this with the group. The way I see it is: RBA's returned value is also pretty much undefined for all the purposes of the user (i.e. the only thing that is defined is that it's not coming from outside of the buffer memory). So if both in-memory and on-stack out-of-bounds accesses are pretty much undefined anyway, why would we prioritize bounds checking (in the spec) one over another?

Reply:
We talked some more about this internally and it's worth raising here. Most languages distinguish in some way between undefined values and undefined behaviour. This would be an undefined value; however, again generally, undefined values can lead to undefined behaviours. For example, most languages treat branching on an undefined value as producing undefined behaviour. So it might well be worth paying a small cost (the clamp/max operation) to prevent that possibility.